fuel/src/main/kotlin/com/github/kittinunf/fuel/core/FuelError.kt

```diff
@@ -1,11 +1,6 @@
 package com.github.kittinunf.fuel.core
 
-import kotlin.properties.Delegates
-
-class FuelError : Exception() {
-    var exception: Exception by Delegates.notNull()
-    var errorData = ByteArray(0)
-    var response = Response()
-
+class FuelError(val exception: Exception, val errorData: ByteArray = ByteArray(0), val response: Response = Response.error())
+    : Exception("FuelError", exception) {
```

Comment:
We should propagate the error so the info could be able to extract by clients. I suggest we use this version of the constructor `class FuelError(..) : Exception(exception)` instead.